clientconn.go

```diff
@@ -779,11 +779,14 @@ func (ac *addrConn) resetTransport(closeTransport bool) error {
 			}
 			ac.mu.Unlock()
 			closeTransport = false
+			timer := time.NewTimer(sleepTime - time.Since(connectTime))
 			select {
-			case <-time.After(sleepTime - time.Since(connectTime)):
+			case <-timer.C:
 			case <-ac.ctx.Done():
+				timer.Stop()
```

Comment:
I think you're right.  I thought the draining of the channel was necessary to unblock a goroutine that was pushing to the timer's channel.  But, it looks like no such goroutine exists, and the channel is buffered anyway.  So this should be fine  as-is.